eng/build.yml

```diff
@@ -0,0 +1,83 @@
+parameters:
+  agentOs: ''
+  queue: {}
+  variables: {}
+
+phases:
+- template: /eng/common/templates/phases/base.yml
+  parameters:
+    name: ${{ parameters.agentOs }}
+    enableTelemetry: true
+    enableMicrobuild: true
+    publicBuildReasons: PullRequest
+    queue: ${{ parameters.queue }}
+    variables: 
+      ${{ insert }}: ${{ parameters.variables }}
+      _AgentOSName: ${{ parameters.agentOs }}
+      _HelixType: build/product
+      _HelixBuildConfig: $(_BuildConfig)
+      BuildConfig: $(_BuildConfig)
+      BuildArchitecture: $(_BuildArchitecture)
+      _HelixSource: pr/dotnet/toolset/$(Build.SourceBranch)
```

Comment:
I'd expect helixsource to be different for public vs internal builds

Reply:
it will be, once we add internal builds. This file only covers public builds.